README.md

```diff
@@ -1,63 +1,109 @@
 
 ## Usage
 
-### Install the features dependencies
+First read the docs about [how to create a new hooks handler](http://dredd.org/en/latest/hooks/new-language.html) for Dredd. Implement your hooks handler and then make sure it passes this test suite.
 
-```bash
-npm install -g dredd
-git clone https://github.com/apiaryio/dredd-hooks-template.git
-cd dredd-hooks-template
-bundle install
-```
+### Installation
+
+1.  Make sure you have [Node.js](https://nodejs.org/) and [npm](https://www.npmjs.com/package/npm) available.
+1.  Create a `package.json` file in the root of your project. This is where your JavaScript dependencies are going to be specified:
+
+    ```json
+    {
+      "scripts": { "test": "dredd-hooks-template test" },
+      "private": true
+    }
+    ```
+
+1.  Run `npm install dredd-hooks-template --save-dev` to install and declare this test suite as your development dependency.
+1.  Run `npx dredd-hooks-template init` to get a copy of the test suite in the `./features` directory.
+1.  Open the feature files in `./features/*.feature` and in all of them
+
+    -   replace `dredd-hooks-{{mylanguage}}` with a path to your hooks handler executable which you want to get tested (e.g. `./bin/dredd_hooks`)
+    -   replace `{{myextension}}` by the extension of the hooks handler language (e.g. `.py`),
+    -   uncomment the code blocks and rewrite them to the hooks handler language.
+
+Now you have the test suite ready.
+
+### Usage
 
-### Enable the features for your language
+Every time you run `npx dredd-hooks-template test` (or `npm test`), you should see the test suite running. The end goal is that all the tests pass:
 
-1. Open the feature files in `./features/*.feature`
-1. In all of them, replace:
-  - `{{mylanguage}}` by the hooks handler command for you language
-  - `{{myextension}}` by the extension for your language
-1. Implement the code examples in your language
-1. Run the test suite: `bundle exec cucumber`
+![test suite passing](passing.png)
+
+You should add the `package.json` file to Git. When starting from scratch, you can run `npm install` to install the JavaScript dependencies.
+
+<a name="upgrading"></a>
+
+### Upgrading
+
+Watch for newer versions of the [dredd-hooks-template package](https://www.npmjs.com/package/dredd-hooks-template) and upgrade regularly to keep up with development of Dredd and the test suite itself. To upgrade, run:
+
+```
+$ npx dredd-hooks-template upgrade
+```
 
-### Add the features to your project
+It upgrades the package to the latest version and copies the latest feature files to the project's `./features/` directory. It won't overwrite the existing files as the names of the new files get suffixed with version. Then it's up to you to compare the old and new files, spot changes, and update the project's test suite.
```

Comment:
We can also use `postinstall` npm hook for copying the files, so users could upgrade the package version using a default flow (`npm outdated`/`npm install ...`), and the hook would handle additional logic.

Also account on the scenario that a user will upgrade the package as usual. I think we should ensure the package would contain expected files in any way.

Reply:
That's an interesting idea which didn't occur to me. One concern I'd have is that it would implicitly, without asking the user, copy some files over from the package to their code. I'm not sure whether it's not too invasive. I kind of like the explicit step user takes - they know what they're doing, they're doing it willingly, they can easily retry, etc.